language-server/src/main/java/org/sonarlint/languageserver/RuleDescriptionHttpServer.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * SonarLint Language Server
+ * Copyright (C) 2009-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarlint.languageserver;
+
+import fi.iki.elonen.NanoHTTPD;
+import fi.iki.elonen.NanoHTTPD.Response.Status;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Locale;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.StringUtils;
+import org.sonarsource.sonarlint.core.client.api.common.RuleDetails;
+import org.sonarsource.sonarlint.core.client.api.standalone.StandaloneSonarLintEngine;
+
+class RuleDescriptionHttpServer extends NanoHTTPD {
+
+  private final StandaloneSonarLintEngine engine;
+  private final String css;
+
+  RuleDescriptionHttpServer(int port, StandaloneSonarLintEngine engine) {
+    super(port);
+    this.engine = engine;
+    try {
+      this.css = IOUtils.toString(this.getClass().getResourceAsStream("/rules.css"), StandardCharsets.UTF_8);
+    } catch (IOException e) {
+      throw new IllegalStateException(e);
+    }
+  }
+
+  @Override
+  public Response serve(IHTTPSession session) {
+    String ruleKey = session.getParms().get("ruleKey");
+    if (ruleKey == null) {
+      return newFixedLengthResponse(Status.BAD_REQUEST, NanoHTTPD.MIME_PLAINTEXT, "Missing 'ruleKey' parameter");
+    } else {
+      RuleDetails ruleDetails;
+      try {
+        ruleDetails = engine.getRuleDetails(ruleKey);
+      } catch (IllegalArgumentException e) {
+        return newFixedLengthResponse(Status.NOT_FOUND, NanoHTTPD.MIME_PLAINTEXT, "No rules with key '" + ruleKey + "'");
+      }
+      try {
+        String ruleName = ruleDetails.getName();
+        String htmlDescription = ruleDetails.getHtmlDescription();
+        String type = ruleDetails.getType().toLowerCase(Locale.ENGLISH);
```

Comment:
But here I'm in standalone mode, so this should never happen. We'll see when introducing connected mode.